### PR TITLE
Feature/implement zustand slices pattern

### LINKED ISSUE #80: Implement Zustand Slices Pattern to reduce BoilerPlate
[Slices Pattern Doc](https://docs.pmnd.rs/zustand/guides/slices-pattern)
[With Typescript](https://docs.pmnd.rs/zustand/guides/typescript#slices-pattern)

### PR DESCRIPTION
**Pull request by tomivm**
In this PR the main store is divided into smaller individual stores using [Zustand Slices Pattern](https://docs.pmnd.rs/zustand/guides/slices-pattern) to achieve modularity and reduce the boilerplate for separate stores. 
With this, we reduce the number of components for providers and avoid composing Providers.
Also, Zustand dev tools are implemented.

Close #80 and #77

### CODE REVIEW

File: src/providers/StoreProvider.tsx [hunk 1]
```diff
@@ -1,15 +1,39 @@
...
+export const StoreContext = createContext<StoreApi<Store> | null>(null);
+
+export default function StoreProvider({ children }: React.PropsWithChildren) {
+  const storeRef = useRef<StoreApi<Store>>();
+  if (!storeRef.current) {
+    storeRef.current = createStore<Store>()(
+      devtools((...props) => ({
+        ...createPromptSlice(...props),
+        ...createBoardSlice(...props),
+      })),
```
Comment (RodriSanchez1) [inline src/providers/StoreProvider.tsx:18]: We can add store name here. 
```javascript
devtools((...props) => ({
        ...createPromptSlice(...props),
        ...createBoardSlice(...props),
      }), { name: 'Cboard-AI-builder' },),

```

**Review by RodriSanchez1 — APPROVED** [review]
Good PR! Congrats! It was really dificult follow the mixed Zustand documentation. We will wait  merge the other 2 open PR until merge this. It will be some conflicts to fix!

File: src/providers/StoreProvider.tsx [hunk 2]
```diff
@@ -1,15 +1,39 @@
...
-    <ComposeComponents components={[PromptStoreProvider, BoardStoreProvider]}>
+    <StoreContext.Provider value={storeRef.current}>
       {children}
-    </ComposeComponents>
+    </StoreContext.Provider>
   );
 }
+
+export const useBoundStore = <StoreProps,>(
+  selector: (store: Store) => StoreProps,
```
Comment (RodriSanchez1) [inline src/providers/StoreProvider.tsx:30]: Leave these comment for the reference [Use Bound Store](https://docs.pmnd.rs/zustand/previous-versions/zustand-v3-create-context#migration)

Comment (tomivm) [reply src/providers/StoreProvider.tsx:30]: What do you mean?

Comment (tomivm) [reply src/providers/StoreProvider.tsx:33]: @RodriSanchez1 Open a new issue to add this documentation by a different PR

**Pull request merged**